lib/src/client.dart

```diff
@@ -121,6 +121,24 @@ class Client extends MatrixApi {
   /// The timeout until a typing indicator gets removed automatically.
   final Duration typingIndicatorTimeout;
 
+  DiscoveryInformation? _wellKnown;
+
+  /// the cached .well-known file updated using [getWellknown]
```

Comment:
```suggestion
  /// the cached .well-known file updated using [getWellknown]
  /// to refresh call getWellknown() again
```

We could also refresh the cache everytime someone calls .wellKnown but do it in a non blocking way.